libclamav/scanners.c

```diff
@@ -2082,6 +2082,452 @@ static cl_error_t cli_ole2_tempdir_scan_for_xlm_and_images(const char *dir, cli_
+    "ms-settings-wifi://",
+    "ms-settings-workplace://",
+    "ms-spd://",
+    "ms-stickers://",
+    "ms-sttoverlay://",
+    "ms-transit-to://",
+    "ms-useractivityset://",
+    "ms-virtualtouchpad://",
+    "ms-visio://",
+    "ms-walk-to://",
+    "ms-whiteboard://",
+    "ms-whiteboard-cmd://",
+    "ms-word://",
+    "msnim://",
+    "msrp://",
+    "msrps://",
+    "mss://",
+    "mt://",
+    "mtqp://",
+    "mumble://",
+    "mupdate://",
+    "mvn://",
+    "mvrp://",
+    "mvrps://",
+    "news://",
+    "nfs://",
+    "ni://",
+    "nih://",
+    "nntp://",
+    "notes://",
+    "num://",
+    "ocf://",
+    "oid://",
+    "onenote://",
+    "onenote-cmd://",
+    "opaquelocktoken://",
+    "openid://",
+    "openpgp4fpr://",
+    "otpauth://",
+    "p1://",
+    "pack://",
+    "palm://",
+    "paparazzi://",
+    "payment://",
+    "payto://",
+    "pkcs11://",
+    "platform://",
+    "pop://",
+    "pres://",
+    "prospero://",
+    "proxy://",
+    "pwid://",
+    "psyc://",
+    "pttp://",
+    "qb://",
+    "query://",
+    "quic-transport://",
+    "redis://",
+    "rediss://",
+    "reload://",
+    "res://",
+    "resource://",
+    "rmi://",
+    "rsync://",
+    "rtmfp://",
+    "rtmp://",
+    "rtsp://",
+    "rtsps://",
+    "rtspu://",
+    "sarif://",
+    "secondlife://",
+    "secret-token://",
+    "service://",
+    "session://",
+    "sftp://",
+    "sgn://",
+    "shc://",
+    "shttp://",
+    "sieve://",
+    "simpleledger://",
+    "simplex://",
+    "sip://",
+    "sips://",
+    "skype://",
+    "smb://",
+    "smp://",
+    "sms://",
+    "smtp://",
+    "snews://",
+    "snmp://",
+    "soap.beep://",
+    "soap.beeps://",
+    "soldat://",
+    "spiffe://",
+    "spotify://",
+    "ssb://",
+    "ssh://",
+    "starknet://",
+    "steam://",
+    "stun://",
+    "stuns://",
+    "submit://",
+    "svn://",
+    "swh://",
+    "swid://",
+    "swidpath://",
+    "tag://",
+    "taler://",
+    "teamspeak://",
+    "tel://",
+    "teliaeid://",
+    "telnet://",
+    "tftp://",
+    "things://",
+    "thismessage://",
+    "tip://",
+    "tn3270://",
+    "tool://",
+    "turn://",
+    "turns://",
+    "tv://",
+    "udp://",
+    "unreal://",
+    "upt://",
+    "urn://",
+    "ut2004://",
+    "uuid-in-package://",
+    "v-event://",
+    "vemmi://",
+    "ventrilo://",
+    "ves://",
+    "videotex://",
+    "vnc://",
+    "view-source://",
+    "vscode://",
+    "vscode-insiders://",
+    "vsls://",
+    "w3://",
+    "wais://",
+    "web3://",
+    "wcr://",
+    "webcal://",
+    "web+ap://",
+    "wifi://",
+    "wpid://",
+    "ws://",
+    "wss://",
+    "wtai://",
+    "wyciwyg://",
+    "xcon://",
+    "xcon-userid://",
+    "xfire://",
+    "xmlrpc.beep://",
+    "xmlrpc.beeps://",
+    "xmpp://",
+    "xftp://",
+    "xrcp://",
+    "xri://",
+    "ymsgr://",
+    "z39.50://",
+    "z39.50r://",
+    "z39.50s://"};
+
+static bool is_url(const char *const str)
```

Comment:
Should we add a `str_len` argument so that it does not have to be NULL-terminated?

Reply:
yeah, makes sense

Reply:
done